api_app/views.py

```diff
@@ -697,9 +697,15 @@ def get_queryset(self):
             user=self.request.user, organization__isnull=False
         )
         if membership.exists():
-            result |= PluginConfig.objects.filter(
-                organization=membership[0].organization,
+            user_membership = membership[0]
+            # only the owner can view the secret
+            extracted = PluginConfig.objects.filter(
+                organization=user_membership.organization,
             )
+            if not user_membership.is_owner:
+                extracted.update(value="redacted")
```

Comment:
You are actually change the value in the db... we do not want that, we just want to change the value returned to the frontend